### ISSUE TITLE
Make FunctionParameterCountRule configurable to support inits

### ISSUE THREAD

**Issue by LipYoung** [realm/SwiftLint#1354]
dear author:
  I use `swiftlint` in my company's project.This is good library.help me and my colleague more.
but, `FunctionParameterCountRule`  don't detected `init` constructor func  in swift 3 and Xcode 8.
  I check source code,it write :
```swift
nonTriggeringExamples: [
            "init(a: Int, b: Int, c: Int, d: Int, e: Int, f: Int) {}",
            ... ...
        ],
```
this is very regrettable.
Want add features.Thinks.

I use English small time.very sorry.Thinks your watch.

**Comment by marcelofabri** [comment]
This is by design: https://github.com/realm/SwiftLint/issues/544

Although I guess we could make it configurable.

**Comment by LipYoung** [comment]
Thanks @marcelofabri .
